File: cpp/problems/problem#2401.cpp
#include <vector>

int longestNiceSubarray(std::vector<int> &nums) {
  int n = nums.size();
  if (n == 1) {
    return 1;
  }
  int l = 0, r = 1;
  int x = nums[0], ans = 1;

  while (r < n) {
    while (l < r && (x & nums[r]) != 0) {
      x ^= nums[l];
      l++;
    }
    x ^= nums[r];
    if (ans < r - l + 1) {
      ans = r - l + 1;
    }
    r++;
  }
  return ans;
}

File: go/problem#2401.go
package main

func longestNiceSubarray(nums []int) int {
	n := len(nums)
	if n == 1 {
		return 1
	}
	l, r := 0, 1
	x := nums[0]
	ans := 1
	for r < n {
		for nums[r]&x != 0 && l < r {
			x ^= nums[l]
			l++
		}
		x ^= nums[r]
		ans = max(ans, r-l+1)
		r++
	}
	return ans
}
